### PR TITLE
Ims provisioning wazi

### PR DESCRIPTION
**Pull request by an-lam**
IMS Provisioning optimized for "vanilla" wazi environment.  I already updated all README files.  The new code is in ims-provisioning-wazi

### CODE REVIEW

File: zos_subsystems/ims/ims_provisioning-wazi/vars_wazi/ims-dbdc.yml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+# DFS_SMS_CLASS: SCNOSMS       # SCNOSMS does not work as expected
+DFS_SMS_CLASS: SCIMS
+# If the volumes are managed by SMS, what SMS management class should be used?
+DFS_SMS_MCLASS: NONE
+
+# -------------------------- volume related settings ------------------------- #
+# ------------------------ for when SMS is not in use ------------------------ #
+
+# first volume to use to store IMS datasets
+DFS_DS_VOLUME1: A5USR1
```
Comment (adn1107) [inline zos_subsystems/ims/ims_provisioning-wazi/vars_wazi/ims-dbdc.yml:97]: Need to find out how to get this value for each ADCD release

File: zos_subsystems/ims/ims_provisioning-wazi/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Inside each role:
+
+* `./tasks` contains tasks that can be performed by the role.
+  Sub-folder names describe the use of the contained files.
+* `./tasks/main.yml` contains the default tasks performed by the role.
+
+
+## Ansible Collection Requirements
+* IBM z/OS core collection 1.4.0
```
Comment (adn1107) [inline zos_subsystems/ims/ims_provisioning-wazi/README.md:63]: should we add 'beta.1' for these collections

**Review by adn1107 — CHANGES_REQUESTED** [review]
Let me know if you have any questions

File: zos_subsystems/ims/ims_provisioning-wazi/inventories/zosendpoint [hunk 3]
```diff
@@ -0,0 +1,7 @@
+all:
+  hosts:
+    zosendpoint:
+      ansible_host:  ansible-zdt11.fyre.ibm.com
```
[the remark this answers is not in the document]
Comment (an-lam) [reply zos_subsystems/ims/ims_provisioning-wazi/inventories/zosendpoint:4]: sure

[the remark this answers is not in the document]
Comment (an-lam) [reply zos_subsystems/ims/ims_provisioning-wazi/inventories/zosendpoint:4]: sure

File: zos_subsystems/ims/ims_provisioning-wazi/README.md (hunk 2 above)
Comment (an-lam) [reply zos_subsystems/ims/ims_provisioning-wazi/README.md:63]: IMS Provisioning works fine with Core collection 1.4.0

File: zos_subsystems/ims/ims_provisioning-wazi/roles/deprovision_ims/meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+galaxy_info:
```
[the remark this answers is not in the document]
Comment (an-lam) [reply zos_subsystems/ims/ims_provisioning-wazi/roles/deprovision_ims/meta/main.yml:1]: this is deprovisioning.  We may not need to provide any credentials.

File: zos_subsystems/ims/ims_provisioning-wazi/README.md (hunk 2 above)
Comment (adn1107) [reply zos_subsystems/ims/ims_provisioning-wazi/README.md:63]: there's no such thing as ibm_zos_core 1.4.0 as see here:
```
ansible-galaxy collection install -U ibm.ibm_zos_core:1.4.0                                       
Starting galaxy collection install process
Process install dependency map
ERROR! Failed to resolve the requested dependencies map. Could not satisfy the following requirements:
* ibm.ibm_zos_core:1.4.0 (direct request)
(venv) (base) ➜  ims_provisioning-wazi git:(ims-provisioning-wazi) ✗ ansible-galaxy collection install -U ibm.ibm_zos_core:1.4.0-beta.1
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Downloading https://galaxy.ansible.com/download/ibm-ibm_zos_core-1.4.0-beta.1.tar.gz to /Users/adn/.ansible/tmp/ansible-local-46955az78w8y6/tmp6njhttp9/ibm-ibm_zos_core-1.4.0-beta.1-4nph_44a
Installing 'ibm.ibm_zos_core:1.4.0-beta.1' to '/Users/adn/.ansible/collections/ansible_collections/ibm/ibm_zos_core'
ibm.ibm_zos_core:1.4.0-beta.1 was installed successfully
```

File: zos_subsystems/ims/ims_provisioning-wazi/README.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+     * ansible_user: the username used to login with SSH.
+     * ansible_python_interpreter: the path on the target to the python interpreter.
+     * ansible_port: the port number to login with SSH if it's not the default port 22.
+   * An example is below, where `zosendpoint` will be the name used to reference our target:
+
+    ```yaml
+    all:
+      hosts:
+        zosendpoint:
+          ansible_host: ec0000a.vmec.svl.ibm.com
```
[the remark this answers is not in the document]
Comment (an-lam) [reply zos_subsystems/ims/ims_provisioning-wazi/README.md:92]: Can we assume all wazis use the same SSH port?

File: zos_subsystems/ims/ims_provisioning-wazi/README.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+          ansible_user: ibmuser
+          ansible_python_interpreter: /usr/lpp/IBM/cyp/v3r9/pyz/bin/python3
+          ansible_port: 2022 
+
+    ```
+
+2. Update the environment variables for the z/OS system in [host_vars/zosendpoint.yml](host_vars/zosendpoint.yml)
```
[the remark this answers is not in the document]
Comment (an-lam) [reply zos_subsystems/ims/ims_provisioning-wazi/README.md:99]: Is SSH port customizable when a customer installs wazi?

File: zos_subsystems/ims/ims_provisioning-wazi/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (adn1107) [reply zos_subsystems/ims/ims_provisioning-wazi/README.md:92]: maybe not. I think the default SSH port is 2022 if they don't customize.

File: zos_subsystems/ims/ims_provisioning-wazi/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (adn1107) [reply zos_subsystems/ims/ims_provisioning-wazi/README.md:99]: i think we need to get the terminology straight. I think Wazi is referring to stuff that in OCP. When not OCP, i think it's ADCD and they use something called zPT tool to bring up the z/os emulator and i would assume this is something they can customize but not sure. To be safe, let's ask the customers to specify this port #.

File: zos_subsystems/ims/ims_provisioning-wazi/vars_wazi/ims-dbdc.yml (hunk 1 above)
Comment (adn1107) [reply zos_subsystems/ims/ims_provisioning-wazi/vars_wazi/ims-dbdc.yml:97]: I have code to set this variable at run-time. Will check in the change after we finish this PR.

File: zos_subsystems/ims/ims_provisioning-wazi/roles/deprovision_ims/meta/main.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (adn1107) [reply zos_subsystems/ims/ims_provisioning-wazi/roles/deprovision_ims/meta/main.yml:1]: I mean the author info

**Comment by adn1107** [comment]
will create a new PR after cleaning up the commit history

**Pull request closed without merging**